packages/react-dom/src/__tests__/ReactServerRenderingHydration-test.js

```diff
@@ -164,52 +203,67 @@ describe('ReactDOMServerHydration', () => {
   });
 
   // Regression test for https://github.com/facebook/react/issues/11726
-  it('should not focus on either server or client with autofocus={false} even if there is a markup mismatch', () => {
+  it('should not focus on either server or client with autofocus={false} even if there is a markup mismatch', async () => {
     const element = document.createElement('div');
     element.innerHTML = ReactDOMServer.renderToString(
       <button autoFocus={false}>server</button>,
     );
     expect(element.firstChild.autofocus).toBe(false);
-
-    element.firstChild.focus = jest.fn();
-
-    expect(() =>
-      ReactDOM.hydrate(<button autoFocus={false}>client</button>, element),
-    ).toErrorDev(
-      'Warning: Text content did not match. Server: "server" Client: "client"',
+    const onFocusBeforeHydration = jest.fn();
+    const onFocusAfterHydration = jest.fn();
+    element.firstChild.focus = onFocusBeforeHydration;
+
+    await expect(async () => {
+      await act(() => {
+        ReactDOMClient.hydrateRoot(
+          element,
+          <button autoFocus={false} onFocus={onFocusAfterHydration}>
+            client
+          </button>,
+          {onRecoverableError: error => {}},
+        );
+      });
+    }).toErrorDev(
+      [
+        'An error occurred during hydration. The server HTML was replaced with client content in <div>.',
+        'Warning: Text content did not match. Server: "server" Client: "client"',
+      ],
+      {withoutStack: 1},
     );
 
-    expect(element.firstChild.focus).not.toHaveBeenCalled();
```

Comment:
We set a focus handler now during hydration so Jest started failing because it received a function that wasn't a mock. I replaced this with dedicated event handlers for before and after hydration that are still not called as intended.